### PR TITLE
http: add codec error response code details

### PR DESCRIPTION
**Pull request by kyessenov**
Signed-off-by: Kuat Yessenov <kuat@google.com>

Description: report protocol error to stream info to be picked up by connection access loggers
Risk Level: low
Testing:
Docs Changes:
Release Notes:
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

File: source/common/http/conn_manager_impl.cc [hunk 1]
```diff
@@ -322,11 +324,6 @@ Network::FilterStatus ConnectionManagerImpl::onData(Buffer::Instance& data, bool
     try {
       codec_->dispatch(data);
     } catch (const FrameFloodException& e) {
-      // TODO(mattklein123): This is an emergency substitute for the lack of connection level
```
Comment (mattklein123) [inline source/common/http/conn_manager_impl.cc]: Can we delete the runtime guarded log below, any associated documentation, etc.?

File: source/common/http/conn_manager_impl.cc [hunk 2]
```diff
@@ -281,6 +281,8 @@ RequestDecoder& ConnectionManagerImpl::newStream(ResponseEncoder& response_encod
 
 void ConnectionManagerImpl::handleCodecException(const char* error) {
   ENVOY_CONN_LOG(debug, "dispatch error: {}", read_callbacks_->connection(), error);
+  read_callbacks_->connection().streamInfo().setResponseCodeDetails(
+      absl::StrCat("codec error: ", error));
```
Comment (mattklein123) [inline source/common/http/conn_manager_impl.cc]: Should we have a response flag here also?

Comment (kyessenov) [reply source/common/http/conn_manager_impl.cc]: Sure, let me take a look.

**Comment by kyessenov** [comment]
Removed the runtime flag and added a response flag to connection stream info. PTAL.

**Comment by kyessenov** [comment]
Updated the version history with an entry.

**Pull request merged**